framework/PostProcessors/ETimporter.py

```diff
@@ -0,0 +1,262 @@
+
+      ## Also, initialize the dictionary of values for this variable so we can
+      ## encode them as integers as well
+      values[event] = []
+
+      ## Iterate through the forks that use this event and gather all of the
+      ## possible states
+      for fork in self.findAllRecursive(eventTree, 'fork'):
+        if fork.get('functional-event') == event:
+          for path in fork.findall('path'):
+            state = path.get('state')
+            if state not in values[event]:
+              values[event].append(state)
+
+    ## Iterate through the sequences and gather all of the possible outcomes
+    ## so we can numerically encode them latter
+    for node in root.findall('define-sequence'):
+      outcome = node.get('name')
+      if outcome not in outcomes:
+        outcomes.append(outcome)
+
+    self.raiseADebug("ETimporter variables identified: " + str(format(self.variables)))
+
+    d = len(self.variables)
+    n = len(self.findAllRecursive(eventTree, 'sequence'))
+
+    self.pointSet = -1 * np.ones((n, d + 1))
+
+    rowCounter = 0
+    for node in eventTree:
+      newRows = self.constructPointDFS(node, self.variables, values, outcomes, self.pointSet, rowCounter)
+      rowCounter += newRows
+
+    outputDict = {}
+    outputDict['inputs']  = {}
+    outputDict['outputs'] = {}
+
+    for index,var in enumerate(self.variables):
+      outputDict['inputs'][var] = self.pointSet[:,index]
+
+    outputDict['outputs']['sequence'] = self.pointSet[:,-1]
+    return outputDict
+
+  def collectOutput(self,finishedJob, output):
+    """
+      Function to place all of the computed data into the output object, (DataObjects)
+      @ In, finishedJob, object, JobHandler object that is in charge of running this postprocessor
+      @ In, output, object, the object where we want to place our computed results
+      @ Out, None
+    """
+    evaluation = finishedJob.getEvaluation()
+    if isinstance(evaluation, Runners.Error):
+      self.raiseAnError(RuntimeError, ' No available output to collect (Run probably is not finished yet) via',self.printTag)
+    if not set(output.getParaKeys('inputs')) == set(self.variables):
+      self.raiseAnError(RuntimeError, ' ETimporter: set of branching variables in the ET ( ' + str(output.getParaKeys('inputs')) + ' ) is not identical to the set of input variables specified in the PointSet (' + str(self.variables) +')')
+
+    outputDict = evaluation[1]
+    # Output to file
+    if output.type in ['PointSet']:
+      for key in output.getParaKeys('inputs'):
+        for value in self.outputDict['inputs'][key]:
+          output.updateInputValue(str(key),value)
+      for key in output.getParaKeys('outputs'):
+        for value in self.outputDict['outputs'][key]:
+          output.updateOutputValue(str(key),value)
+    else:
+      self.raiseAWarning('Output type ' + str(output.type) + ' is not supported.')
+
+  def findAllRecursive(self, node, element, result = None):
+    """
```

Comment:
I think the same outcome is achievable with the ```Element.findall()``` method

Reply:
this method differs from findall() since it goes for all nodes, subnodes, subsubnodes etc. recursively

Reply:
Sorry my bad... It is not find all but ```iter```
For example, 
```
import xml.etree.ElementTree as ET
results = node.iter(element)
```

Reply:
done!